launchflow/gcp/firebase_site.py

```diff
@@ -0,0 +1,108 @@
+from dataclasses import dataclass
+from typing import List, Optional
+
+from launchflow import exceptions
+from launchflow.gcp.firebase import FirebaseHostingSite, FirebaseProject
+from launchflow.gcp.service import GCPStaticService
+from launchflow.models.enums import ServiceProduct
+from launchflow.node import Inputs
+from launchflow.resource import Resource
+from launchflow.service import DNSOutputs, DNSRecord, StaticServiceOutputs
+
+
+@dataclass
+class FirebaseStaticSiteInputs(Inputs):
+    pass
+
+
+class FirebaseStaticSite(GCPStaticService):
+    """A service hosted on Firebase Hosting.
+
+    ### Example Usage
+    ```python
+    import launchflow as lf
+
+    website = lf.gcp.FirebaseStaticSite("my-website", static_directory="path/to/local/files")
+    ```
+    """
+
+    product = ServiceProduct.GCP_FIREBASE_STATIC_SITE.value
+
+    def __init__(
+        self,
+        name: str,
+        # static inputs
+        static_directory: str,
+        *,
+        static_ignore: List[str] = [],
+        # backend bucket inputs
+        region: Optional[str] = None,
+        domain: Optional[str] = None,
+    ) -> None:
+        """Creates a new Cloud Run service.
+
+        **Args:**
+        - `name (str)`: The name of the service.
+        - `static_directory (str)`: The directory of static files to serve. This should be a relative path from the project root where your `launchflow.yaml` is defined.
+        - `static_ignore (List[str])`: A list of files to ignore when deploying the service. This can be in the same syntax you would use for a `.gitignore`.
+        - `region (Optional[str])`: The region to deploy the service to.
+        - `domain (Optional[str])`: The custom domain to map to the service.
+        """
+        super().__init__(
+            name=name,
+            static_directory=static_directory,
+            static_ignore=static_ignore,
+        )
+        self.region = region
+        self.domain = domain
+
+        # TODO: make this configurable
```

Comment:
mind filing an issue for this so we can keep track of it? will also be good to keep context around what we want to be configurable.

Of the top of my head probably want to allow users to create a firebase project out side of this

Reply:
There's honestly nothing to really configure with firebase, its basically like enabling a service on the gcp project. This is just a reminder for me to include the static dir configuration in my next pass